File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    return sum(range(count + 1)) ** 2


def sum_of_squares(count):
    return sum(map(lambda x: x ** 2, range(count + 1)))
    # return sum([x ** 2 for x in range(count + 1)])


def difference(count):
    return square_of_sum(count) - sum_of_squares(count)

# vim:ts=2:sw=2:expandtab

File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares solves diffsquares.
package diffsquares

// SumOfSquares sums up a sequence of squares.
func SumOfSquares(n int) int {
	var sum int
	for i := 1; i <= n; i++ {
		sum += i * i
	}
	return sum
}

// SquareOfSum is the square of the sum of a sequence.
func SquareOfSum(n int) int {
	var sum int
	for i := 1; i <= n; i++ {
		sum += i
	}
	return sum * sum
}

// Difference between SquareOfSum and SumOfSquares.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
